### PR TITLE
Host components doc

### PR DESCRIPTION
**Pull request by vitek-karas**
Add at least some documentation of the various components (mostly libraries) which are involved in .NET Core hosting and how they interact.

### CODE REVIEW

**Comment by vitek-karas** [comment]
@steveharter if you know of anybody else who has experience in this area, could you please add them to the PR?

File: Documentation/design-docs/host-components.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The .NET Core default hosting setup consists of several components which are described here.
+
+## Executable
+The executable, which is also sometimes refered to as the "host" can be either the
+* `dotnet.exe` - which comes from a shared location and is typically the latest version available on the machine
+* `apphost.exe` - which is used in case of self-contained apps. The name of the .exe will be app dependent and it comes with the app
+
+The executable does just one thing, it finds the `hostfxr.dll` (for Windows, Linux and Max use their respective naming schemes) and passes control to it.
+For `dotnet.exe` host the hostfxr is located next to the host.
```
Comment (steveharter) [inline Documentation/design-docs/host-components.md]: For dotnet.exe hostfxr is obtained from the .\shared\host\fxr\<highestversion>

File: Documentation/design-docs/host-components.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+For self-contained apps (apphost scenario):
+1. The app's folder is searched first
```
Comment (steveharter) [inline Documentation/design-docs/host-components.md]: Note that there can be a relative path also embedded in the apphost.exe

File: Documentation/design-docs/host-components.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+This library finds and resolves the runtime and all the frameworks the app needs. Then it loads the `hostpolicy.dll` (on Windows) and transfers control to it.
+
+The library reads the `.runtimeconfig.json` of the app (and all it's dependent frameworks) and resolves the frameworks. It implements the algorithm for framework resolution as described in [SharedFX Lookup](multilevel-sharedfx-lookup.md).
+
+In most cases the hostfxr used is the latest version available on the machine (self-contained apps use the one they bring with them).
+
+The main reason to split the executable host and the hostfxr is to allow for servicing the logic in hostfxr without the need to stop all instances of the executable host currently running.
+
+## Host Policy
+The host policy library implements all the policies to actually load the runtime, apply configuration, resolve all app's dependencies and actually run the app.
```
Comment (steveharter) [inline Documentation/design-docs/host-components.md]: Probably want something like "calls the CLR with the configuration information which runs the app"

**Pull request merged**